### ISSUE TITLE
Allow calendar to be used without a startDate

### ISSUE THREAD

**Issue by gareth-ib** [krystalcampioni/vue-hotel-datepicker#91]
### Description

The docs on how to use endDate are using the startDate field, so there is no docs to go by.
My goal is to set anything after todays date as disabled, everything before and including today to be enabled.

### Code sample 
this sets anything before todays date as disabled...
```html
	<HotelDatePicker
	    :endDate="new Date(new Date().getFullYear(), 11, 31)"
	></HotelDatePicker>
```
This sets everything before the startDate as disabled; everything after startDate is enabled, including past the endDate
```html
	<HotelDatePicker
	    :startDate="new Date(new Date().getFullYear(), 1, 31)"
	    :endDate="new Date(new Date().getFullYear(), 11, 31)"
	></HotelDatePicker>
```
this errors
`[Vue warn]: Error in beforeMount hook: "Error: Invalid Date in fecha.format"`
```html
	<HotelDatePicker
	    endDate="new Date(new Date().getFullYear(), 11, 31)"
	></HotelDatePicker>
```

### Datepicker Version
vue-hotel-datepicker@2.2.7

**Comment by amulchinock** [comment]
@krystalcampioni Based on some experimentation I've just done - it looks like the datepicker assumes we don't want to book dates in the past (as per @gareth-ib use case).

Is this a bug or a feature?

On the one hand, it makes sense that we would assume no one would want to book past dates (you know, booking hotels and all....)

But, on the other, as @gareth-ib describes - we have the `startDate` and `endDate` parameters, which should (in theory) override this behaviour.

How should we treat this - feature request, or bug fix?

**Comment by krystalcampioni** [comment]
### TLDR ###
With the current implementation, it's not possible to use the calendar without a startDate, as the value is used to generate the months when we navigate on the calendar. You could set a date in the past as the startDate, but this would cause the calendar to open on that date in the past, instead of the current month. 
In the example bellow, the calendar would open initially in 01/01/1970

```
      <DatePicker
        :showYear="true"
        startDate="01/01/1970"
      />

```
### Context ###
Indeed the datepicker was built aiming at hotel booking, thus past dates would not be allowed. 
Initially, the `startDate` wasn't a prop, but an internal variable that was used to create the initial month calendars once we open the datepicker. After a while, it was decided to allow this value to be set as a prop to make the component more flexible.

To give a bit more context, the current implementation works as follows:
- The `startDate` is used to generate 2 months: the month of the startDate and the next one. 
- Once we navigate to the following months, we generate more months and allow the user to navigate back, but only up to the month of the start date.

With that in mind, it's not currently possible to allow _any_ date in the past to be available, as the months are generated always forward, not backwards.

This is an interesting functionality though, and it might make sense to revisit it in the future. I'll mark it as a feature request 👍

**Comment by krystalcampioni** [comment]
@gareth-ib regarding your last example, the error is caused due to the lack of the v-bind colon at the beginning of the prop name. Without it, it treats the value you're passing as a string. It should be:
`:endDate="new Date(new Date().getFullYear(), 11, 31)"`

**Comment by matiasperrone** [comment]
@gareth-ib please try the v4 version. Is in beta but soon to be realeased
